src/components/Modal.mock.ts

```diff
@@ -0,0 +1,29 @@
+import { FunctionComponent } from 'react'
+
+
+export type ModalExports = typeof import('src/components/Modal') & { __esModule: true }
+
+/**
+ * provides a mocked version of Modal module's Modal component that avoids
+ * pitfalls specific to missing services in jest simulated browser environment
+ * including dom measurement calls.
+ */
+export const mockModalModule = (): ModalExports => {
+  const originalModule = jest.requireActual<ModalExports>('src/components/Modal')
+
+  type ModalFn = FunctionComponent & {propTypes: typeof originalModule.modalPropTypes};
```

Comment:
What does this syntax do?

Reply:
FunctionComponent is helpful typing from React, but defines propTypes as optional and non-specific.  To satisfy the type checks on ModalExports, propTypes needs to not be optional and match with the specific propTypes defined in Modal.js

Reply:
Tangential: I'm not sure if I interpret the syntax with `(): ModalExports` correctly. Does it mean `mockModalModule` takes no input and outputs a value of type `ModalExports`?

Reply:
correct.  Depending on the exact situation, Typescript uses : or => to indicate function return type is to follow.

`(arg: string) => number`

is the same as 

`(arg: string): number`

... but one or the other is required to avoid confusion with Javascript mechanics (is my assumption) depending on if you are decorating a lambda/arrow-function or something with a this context (class method)... and some other scenarios.  To be honest, I try one... then the other if first try failed. LOL.

Reply:
In a TypeScript future, does this line becomes simpler?
```ts
type ModalFn = ModalExports['Modal']
```

Reply:
note tho that the desired modal fn export is the default export.  To make sure I am reading your code snippet of the future correctly, you are assuming Modal export is a defined type?  Or did you mean
`type ModalFn = typeof ModalExports['Modal']`
assuming that Modal is a named export and not the default export anymore?

Reply:
> Or did you mean type ModalFn = typeof ModalExports['Modal'] assuming that Modal is a named export and not the default export anymore?

Yes. I wanted to point out that some of the complexity here is due to the current implementation of Modal and, if we discourage default exports and replace prop types with TS, then generally this pattern will be simpler in the future.